src/test/groovy/edu/ie3/datamodel/io/sink/SqlSinkTest.groovy

```diff
@@ -0,0 +1,402 @@
+import edu.ie3.test.common.TimeSeriesTestData
+import edu.ie3.test.helper.TestContainerHelper
+import edu.ie3.util.TimeUtil
+import edu.ie3.util.io.FileIOUtils
+import org.jetbrains.annotations.NotNull
+import org.testcontainers.containers.Container
+import org.testcontainers.containers.PostgreSQLContainer
+import org.testcontainers.utility.MountableFile
+import spock.lang.Shared
+import spock.lang.Specification
+import org.testcontainers.spock.Testcontainers
+import tech.units.indriya.quantity.Quantities
+
+import javax.measure.Quantity
+import javax.measure.quantity.Power
+import java.nio.file.Path
+
+import edu.ie3.test.common.SystemParticipantTestData
+
+import java.sql.SQLException
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+
+import static edu.ie3.datamodel.models.StandardUnits.ENERGY_PRICE
+import static edu.ie3.util.quantities.PowerSystemUnits.DEGREE_GEOM
+import static edu.ie3.util.quantities.PowerSystemUnits.DEGREE_GEOM
+import static edu.ie3.util.quantities.PowerSystemUnits.KILOVOLTAMPERE
+import static tech.units.indriya.unit.Units.PERCENT
+
+@Testcontainers
+class SqlSinkTest extends Specification implements TestContainerHelper, TimeSeriesTestData {
+
+    @Shared
+    PostgreSQLContainer postgreSQLContainer = new PostgreSQLContainer("postgres:14.2")
+
+    @Shared
+    SqlConnector connector
+
+    @Shared
+    SqlDataSource sqlSource
+
+    @Shared
+    DatabaseNamingStrategy namingStrategy
+
+    @Shared
+    DatabaseIdentifier identifier
+
+    static String schemaName = "public"
+
+    def setupSpec() {
+        // Copy sql import scripts into docker
+        MountableFile sqlImportFile = getMountableFile("_sql/")
+        postgreSQLContainer.copyFileToContainer(sqlImportFile, "/home/")
+        postgreSQLContainer.execInContainer("psql", "-Utest", "-f/home/" + "setup.sql")
+
+        connector = new SqlConnector(postgreSQLContainer.jdbcUrl, postgreSQLContainer.username, postgreSQLContainer.password)
+
+        namingStrategy = new DatabaseNamingStrategy()
+
+        identifier = new DatabaseIdentifier("vn_simona", UUID.fromString("8e6bd444-4580-11ee-be56-0242ac120002"))
+
+        sqlSource = new SqlDataSource(connector, schemaName, namingStrategy)
+    }
+
+    def setup() {
+        // Execute import script
+        Iterable<String> importFiles = Arrays.asList(
+                "types.sql",
+                "result_entities.sql",
+                "input_entities.sql",
+                "time_series.sql",
+                "load_profile.sql"
+        )
+        for (String file: importFiles) {
+            Container.ExecResult res = postgreSQLContainer.execInContainer("psql", "-Utest", "-f/home/" + file)
+            assert res.stderr.empty
+        }
+    }
+
+    def cleanup() {
+        postgreSQLContainer.execInContainer("psql", "-Utest", "-f/home/" + "cleanup.sql")
+    }
+
+    def "SQL sink can persist provided elements correctly"() {
```

Comment:
Maybe there are some further tests necessary?

Reply:
I think that looks quite good for now. What I would strongly advocate for is doing some IT test after the source is implemented, to see if the JGC that we persisted comes out the same way we put it in.